### ISSUE TITLE
Method with name `eval` is always reported as error

### ISSUE THREAD

**Issue by Philosoft** [squizlabs/PHP_CodeSniffer#3607]
**Describe the bug**

PHPCS considers methods with name `eval` to be incorrect

**Code sample**

```php
<?php

declare(strict_types=1);

namespace App;

class Foo
{
    public function eval()
    {
        return;
    }
}
```

**Custom ruleset**
No custom rules. Just PSR12 standrad

**To reproduce**

1. Create a file called `test.php` with the code sample above
2. Run `phpcs --standard=PSR12 test.php`
3. See error message displayed

```
----------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
----------------------------------------------------------------------
 9 | ERROR | Expected "function abc(...)"; found "function abc(...)"
----------------------------------------------------------------------
```

If you change method name to basically anything else - `eval1`, `evaluate` etc - there is no error

**Expected behavior**
No complains.

**Versions (please complete the following information):**
 - OS: Ubuntu 20.04
 - PHP: 7.4.3
 - PHPCS: master (5e27199)
 - Standard: PSR12

Same thing happens with PHP 8.1.6

**Comment by jrfnl** [comment]
Thanks for reporting this @Philosoft !

I've confirmed the bug.

The `eval` keyword in a function declaration tokenized (correctly) as a `T_STRING` in PHPCS 3.6.2.
The change in #3484 to improve context sensitive keyword tokenization did not account for the `eval` keyword and broke this.

PR #3608 should fix this. Testing appreciated.

**Comment by gsherwood** [comment]
Thanks for finding and reporting this before it was released

**Comment by Mitiaj** [comment]
Same issue with methods named `isSet($args)` and `empty($args)`

**Comment by gsherwood** [comment]
@Mitiaj Those were also fixed in the same PR created by @jrfnl
